fili-core/src/test/groovy/com/yahoo/bard/webservice/util/IntervalUtilsSpec.groovy

```diff
@@ -326,6 +328,32 @@ class IntervalUtilsSpec extends Specification {
         DAY   | ["2012-02-03/2017-04-04"] | ["2012-02-03/2012-05-04", "2017-02-03/2017-04-04"]
     }
 
+    @Unroll
+    def "getCoarsestTimeGrain returns duration of #expected among #duration1, #duration2, #duration3"() {
+        given:
+        ZonedTimeGrain zonedTimeGrain1 = Mock(ZonedTimeGrain)
+        ZonedTimeGrain zonedTimeGrain2 = Mock(ZonedTimeGrain)
+        ZonedTimeGrain zonedTimeGrain3 = Mock(ZonedTimeGrain)
+
+        zonedTimeGrain1.getEstimatedDuration() >> new Duration(duration1)
+        zonedTimeGrain2.getEstimatedDuration() >> new Duration(duration2)
+        zonedTimeGrain3.getEstimatedDuration() >> new Duration(duration3)
+
+        expect:
+        IntervalUtils.getCoarsestTimeGrain([zonedTimeGrain1, zonedTimeGrain2, zonedTimeGrain3]).getEstimatedDuration().equals(new Duration(expected))
+
+        where:
+        duration1 | duration2 | duration3 | expected
+        1         | 2         | 3         | 1
+        1         | 1         | 3         | 1
+        3         | 3         | 3         | 3
```

Comment:
Any reason we're using mocked grains instead of using real grains and giving them a timezone?